src/main/java/com/gmail/yuramitryahin/security/UserDetailsServiceImp.java

```diff
@@ -0,0 +1,42 @@
+package com.gmail.yuramitryahin.security;
+
+import static org.springframework.security.core.userdetails.User.withUsername;
+
+import com.gmail.yuramitryahin.model.Role;
+import com.gmail.yuramitryahin.model.User;
+import com.gmail.yuramitryahin.service.UserService;
+import java.util.Set;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.User.UserBuilder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Component;
+
+@Component
+public class UserDetailsServiceImp implements UserDetailsService {
+    private final UserService userService;
+
+    @Autowired
+    public UserDetailsServiceImp(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        User user = userService.findByEmail(username).orElseThrow(() ->
+                new UsernameNotFoundException("User not found!"));
```

Comment:
add method input param to make message more informative

Reply:
fixed